types/figlet/index.d.ts

```diff
@@ -0,0 +1,423 @@
+        | 'Shimrod'
+        | 'Short'
+        | 'Slant Relief'
+        | 'Slant'
+        | 'Slide'
+        | 'Small Caps'
+        | 'Small Isometric1'
+        | 'Small Keyboard'
+        | 'Small Poison'
+        | 'Small Script'
+        | 'Small Shadow'
+        | 'Small Slant'
+        | 'Small Tengwar'
+        | 'Small'
+        | 'Soft'
+        | 'Speed'
+        | 'Spliff'
+        | 'Stacey'
+        | 'Stampate'
+        | 'Stampatello'
+        | 'Standard'
+        | 'Star Strips'
+        | 'Star Wars'
+        | 'Stellar'
+        | 'Stforek'
+        | 'Stick Letters'
+        | 'Stop'
+        | 'Straight'
+        | 'Stronger Than All'
+        | 'Sub-Zero'
+        | 'Swamp Land'
+        | 'Swan'
+        | 'Sweet'
+        | 'THIS'
+        | 'Tanja'
+        | 'Tengwar'
+        | 'Term'
+        | 'Test1'
+        | 'The Edge'
+        | 'Thick'
+        | 'Thin'
+        | 'Thorned'
+        | 'Three Point'
+        | 'Ticks Slant'
+        | 'Ticks'
+        | 'Tiles'
+        | 'Tinker-Toy'
+        | 'Tombstone'
+        | 'Train'
+        | 'Trek'
+        | 'Tsalagi'
+        | 'Tubular'
+        | 'Twisted'
+        | 'Two Point'
+        | 'USA Flag'
+        | 'Univers'
+        | 'Varsity'
+        | 'Wavy'
+        | 'Weird'
+        | 'Wet Letter'
+        | 'Whimsy'
+        | 'Wow'
+        ;
+
+    type KerningMethods =
+        | 'default'
+        | 'full'
+        | 'fitted'
+        | 'controlled smushing'
+        | 'universal smushing'
+        ;
+    type PrintDirection = number;
+
+    interface FittingRules {
+        vLayout: number;
+        vRule5: boolean;
+        vRule4: boolean;
+        vRule3: boolean;
+        vRule2: boolean;
+        vRule1: boolean;
+        hLayout: number;
+        hRule6: boolean;
+        hRule5: boolean;
+        hRule4: boolean;
+        hRule3: boolean;
+        hRule2: boolean;
+        hRule1: boolean;
+    }
+
+    interface Options {
+        font?: Fonts;
+        horizontalLayout?: KerningMethods;
+        verticalLayout?: KerningMethods;
+        printDirection?: PrintDirection;
+        showHardBlanks?: boolean;
+    }
+    interface FontOptions {
+        hardBlank: string;
+        height: number;
+        baseline: number;
+        maxLength: number;
+        oldLayout: number;
+        numCommentLines: number;
+        printDirection: PrintDirection;
+        fullLayout: number | null;
+        codeTagCount: number | null;
+        fittingRules: FittingRules;
+    }
+    interface Defaults {
+        font: Fonts;
+        fontPath: string;
+    }
+
+    interface TextMethod {
+        (
+            txt: string,
+            cb: (error: Error | null, result?: string) => void,
+        ): void;
+        (
+            txt: string,
+            font: Fonts,
+            cb: (error: Error | null, result?: string) => void,
+        ): void;
+        (
+            txt: string,
+            // tslint:disable-next-line: unified-signatures
```

Comment:
Combine  signatures and do not disable lint warning

Reply:
@sheetalkamat But `Fonts` type is too long, so it makes hard to read the signatures in IDE/Editors....

Reply:
I guess this is OK for now. I've added Microsoft/TypeScript#26106 to track this.